specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json

```diff
@@ -5330,6 +5539,76 @@
           "description": "Error when What-If operation fails."
         }
       }
+    },
+    "Tags": {
+      "description": "key and value pairs for tags",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
+            "description": "The tag value."
+          }
+        }
+      }
+    },
+    "TagPatchRequest": {
+      "type": "object",
+      "description": "Tag Request for Patch operation.",
+      "properties": {
+        "operation": {
```

Comment:
this I think should be generalized. We should create an Azure extension (similar to x-ms-long-running). and then any service who want to support advanced patch capabilities can leverage that extension. something like - "x-ms-patch-mode"
@anthony-c-martin @KrisBash - thoughts?

Reply:
Since the request body for put and patch is different here, we have defined two entities: TagPatchRequest and TagRequest.